lib/srv/db/db_test.go

```diff
@@ -283,6 +284,72 @@ func TestMySQLAccess(t *testing.T) {
 	}
 }
 
+type testModules struct {
+	modules.Modules
+}
+
+func (m *testModules) Features() modules.Features {
+	return modules.Features{
+		DB: false, // Explicily turn off database access.
+	}
+}
+
+// TestDatabaseAccessDisabled makes sure database access can be disabled via
+// modules.
+func TestDatabaseAccessDisabled(t *testing.T) {
+	defaultModules := modules.GetModules()
+	defer modules.SetModules(defaultModules)
+	modules.SetModules(&testModules{})
+
+	ctx := context.Background()
+	testCtx := setupTestContext(ctx, t)
+	t.Cleanup(func() { testCtx.Close() })
+
+	// Start multiplexer.
+	go testCtx.mux.Serve()
+	// Start fake Postgres server.
+	go testCtx.postgresServer.Serve()
+	// Start database proxy server.
+	go testCtx.proxyServer.Serve(testCtx.mux.DB())
+	// Start database service server.
+	go func() {
+		for conn := range testCtx.proxyConn {
+			testCtx.server.HandleConnection(conn)
+		}
+	}()
+
+	userName := "alice"
+	roleName := "admin"
+	dbUser := "postgres"
+	dbName := "postgres"
+
+	// Create user/role with the requested permissions.
+	_, role, err := auth.CreateUserAndRole(testCtx.tlsServer.Auth(), userName, []string{roleName})
+	require.NoError(t, err)
+
+	role.SetDatabaseNames(services.Allow, []string{types.Wildcard})
```

Comment:
```suggestion
	role.SetDatabaseNames(types.Allow, []string{types.Wildcard})
```
use the new package in new code.